src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,532 @@
+        # Import qpax just in this method
+        import qpax
+
+        # Compute kin-dyn quantities used in the contact model
+        with data.switch_velocity_representation(VelRepr.Mixed):
+            M = js.model.free_floating_mass_matrix(model=model, data=data)
+            J_WC = js.contact.jacobian(model=model, data=data)
+            W_H_C = js.contact.transforms(model=model, data=data)
+        terrain_height = jax.vmap(self.terrain.height)(position[:, 0], position[:, 1])
+        terrain_normal = jax.vmap(self.terrain.normal)(position[:, 0], position[:, 1])
+        n_collidable_points = model.kin_dyn_parameters.contact_parameters.point.shape[0]
+
+        # Compute the activation state of the collidable points
+        inactive_collidable_points, (delta, delta_dot) = RigidContacts._detect_contacts(
+            W_o_C=position,
+            W_o_dot_C=velocity,
+            terrain_height=terrain_height,
+            terrain_normal=terrain_normal,
+        )
+
+        delassus_matrix = RigidContacts._delassus_matrix(M=M, J_WC=J_WC)
+        # Make it symmetric if not
+        delassus_matrix = jax.lax.cond(
+            jnp.allclose(delassus_matrix, delassus_matrix.T),
+            lambda G: G,
+            lambda G: (G + G.T) / 2,
+            delassus_matrix,
+        )
+        # Add regularization for better numerical conditioning
+        delassus_matrix = delassus_matrix + 1e-6 * jnp.eye(delassus_matrix.shape[0])
+
+        nu_dot_free_mixed = RigidContacts._compute_mixed_nu_dot_free(
+            model, data, references=None
+        )
+
+        free_contact_acc = RigidContacts._linear_acceleration_of_collidable_points(
+            model,
+            data,
+            nu_dot_free_mixed,
+        ).flatten()
+        # Compute stabilization term
+        baumgarte_term = RigidContacts._compute_baumgarte_stabilization_term(
+            inactive_collidable_points=inactive_collidable_points,
+            delta=delta,
+            delta_dot=delta_dot,
+            K=self.parameters.K,
+            D=self.parameters.D,
+        ).flatten()
+        free_contact_acc -= baumgarte_term
+
+        # Setup optimization problem
+        Q = delassus_matrix
+        q = free_contact_acc
+        G = RigidContacts._compute_ineq_constraint_matrix(
+            inactive_collidable_points=inactive_collidable_points, mu=self.parameters.mu
+        )
+        h = RigidContacts._compute_ineq_bounds(n_collidable_points=n_collidable_points)
+        A = jnp.zeros((0, 3 * n_collidable_points))
+        b = jnp.zeros((0,))
+
+        jax.debug.print(
+            "Shapes: Q={Q}, q={q}, A={A}, b={b}, G={G}, h={h}",
+            Q=Q.shape,
+            q=q.shape,
+            A=A.shape,
+            b=b.shape,
+            G=G.shape,
+            h=h.shape,
+        )
+
+        # Solve the optimization problem
+        solution, s, z, y, converged, iters = qpax.solve_qp(  # noqa: F841
+            Q=Q, q=q, A=A, b=b, G=G, h=h
+        )
```

Comment:
Just a reminder to remove the unused arguments when the debugging phase is completed
```suggestion
        solution, *_ = qpax.solve_qp(  # noqa: F841
            Q=Q, q=q, A=A, b=b, G=G, h=h
        )
```